src/crypto/transactionUtils.js

```diff
@@ -56,6 +59,32 @@ export const processTxHistoryData = (
   const isIntraWallet = hasOnlyOwnInputs && hasOnlyOwnOutputs
   const isMultiParty =
     ownInputs.length > 0 && ownInputs.length !== tx.inputs.length
+
+  // check if tx includes a deregistration (which means output may contain
+  // a deposit to this wallet)
+  const hasKeyDeregistration = (() => {
+    let _hasKeyDeregistration = false
+    for (const cert of tx.certificates) {
+      if (
+        cert.kind === CERTIFICATE_KIND.STAKE_DEREGISTRATION &&
```

Comment:
Pool retirement also has a refund FYI.

The way you're implementing this function also differs from the Yoroi Extension. I suggest you look at the call to `getFromUserPerspective` in `updateTransactions.js` in the Yoroi Extension along with related comments.

Reply:
Thanks, I'll take a look 👍 

Also note that we have a difference in the way amounts are computed. The extension adds the fee to the amount while mobile doesn't.

Reply:
@SebastienGllmt I refactored the code and tried to match it to the Yoroi extension (last 2 commits).

Also all other comments have been addressed.